toffy/panel_utils.py

```diff
@@ -0,0 +1,106 @@
+import os
+import pandas as pd
+
+from pathlib import Path
+
+from ark.utils import io_utils
+
+
+def convert_panel(panel_path):
+    """
+    Converts the panel retrieved from ionpath into a necessary toffy format,
+    also adds necessary channels for analysis
+    Args:
+        panel_path (str): direct path to panel file
+    Returns:
+        panel (pd.DataFrame): detailing the Mass, Target, Start, and Stop values
+        also saves the above to file with name formatted "panel_name-toffy.csv"
+    """
+
+    # retrieve the panel name and directory it is contained in
+    panel_name = os.path.basename(panel_path).split('.')[0]
+    panel_dir = os.path.dirname(panel_path)
+
+    # read in the example_panel_file to have necessary additional targets
+    example_panel = pd.read_csv(os.path.join(Path(__file__).parent.parent, 'files',
+                                             'example_panel_file.csv'), index_col=False)
+
+    # read in metadata contained in the first few lines
+    with open(panel_path) as r:
+        lines = r.readlines()
+        for index, line in enumerate(lines):
+            # line containing the column names we want
+            if len(line.split(',')) == 10:
+                col_index = index
+                break
+        r.close()
+
+    # load in the panel info to df
+    with open(panel_path) as r:
+        headers = [next(r) for i in list(range(col_index+1))]
+        cols = headers[col_index].split(',')
+        panel = pd.read_csv(r, sep=',', names=cols, index_col=False)
+        r.close()
+
+    # retrieve column names, and original mass / target values
+    panel.columns = [panel.columns.str.replace('"', '')]
+    panel.columns = panel.columns.get_level_values(0)
+    toffy_panel = panel[['Mass', 'Target']].copy()
+
+    mass_start = []
+    mass_stop = []
+
+    for i, row in toffy_panel.iterrows():
```

Comment:
In general, it's better to do one thing at a time. This code is kind of hard to figure out because it's both removing duplicates and creating the `start` and `stop` intervals. 

Instead, I think it would be better to do the following:
1. Find all duplicate mass values in the panel
2. For the duplicates of each mass (there could be more than 1), concatenate them together. This can probably be a helper function that takes in a df and returns a deduped one
3. Generate the mass_start and mass_stop with a vector operation across the entire df at once,  now that we know its clean